pages/settings/wallets/ConnectWallet.tsx

```diff
@@ -0,0 +1,281 @@
+  const [redirectCountdown, setRedirectCountdown] = React.useState<
+    number | null
+  >(null);
+
+  const leftPlugAnim = React.useRef(new Animated.Value(0)).current;
+  const rightPlugAnim = React.useRef(new Animated.Value(0)).current;
+
+  React.useEffect(() => {
+    if (nwcUrl) {
+      Animated.parallel([
+        Animated.timing(leftPlugAnim, {
+          toValue: 5,
+          duration: 500,
+          useNativeDriver: true,
+          easing: Easing.ease,
+        }),
+        Animated.timing(rightPlugAnim, {
+          toValue: -15,
+          duration: 500,
+          useNativeDriver: true,
+          easing: Easing.ease,
+        }),
+      ]).start();
+    } else {
+      leftPlugAnim.setValue(0);
+      rightPlugAnim.setValue(0);
+    }
+  }, [nwcUrl, leftPlugAnim, rightPlugAnim]);
+
+  React.useEffect(() => {
+    if (redirectCountdown === null || !nwcUrl) {
+      return;
+    }
+
+    if (redirectCountdown === 0) {
+      (async () => {
+        const callbackUrl = `${callback}?value=${nwcUrl}`;
+        try {
+          await openURL(callbackUrl);
+        } catch (error) {
+          console.error(error);
+          errorToast(
+            new Error("Couldn't open URL, do you have the app installed?"),
+          );
+        }
+        if (router.canDismiss()) {
+          router.dismissAll();
+        }
+        router.replace("/");
+      })();
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setRedirectCountdown((prev) => (prev ? prev - 1 : prev));
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [redirectCountdown, nwcUrl, callback]);
+
+  const confirm = async () => {
+    setLoading(true);
+    try {
+      const nwcClient = useAppStore.getState().nwcClient;
+      if (!nwcClient) {
+        throw new Error("NWC client not connected");
+      }
+      const secretKey = generateSecretKey();
+      const pubkey = getPublicKey(secretKey);
+      {
+        /* TODO: REPLACE WITH BUDGET INFO (AND METHODS?) */
+      }
+      const response = await nwcClient.createConnection({
+        pubkey,
+        name: appname,
+        budget: {
+          budget: 100_000,
+          renewal_period: "monthly",
+        },
+        methods: [
+          "get_info",
+          "get_balance",
+          "get_budget",
+          "make_invoice",
+          "pay_invoice",
+          "lookup_invoice",
+          "list_transactions",
+          "sign_message",
+        ],
+        metadata: null,
+      });
+
+      console.info("createConnection response", response);
+
+      const newUrl = `nostr+walletconnect://${response.wallet_pubkey}?secret=${bytesToHex(
+        secretKey,
+      )}&relay=${nwcClient.relayUrl}`;
+
+      setNwcUrl(newUrl);
+      setRedirectCountdown(5);
+    } catch (error) {
+      console.error(error);
+      errorToast(error);
+    }
+    setLoading(false);
+  };
+
+  return (
+    <>
+      <Screen
+        title={nwcUrl ? "Wallet Connected" : "Connect Wallet"}
```

Comment:
can we split this into 2 screens?

Reply:
We decided to split the components instead.